media/admin/com_virtuemart/models/zasilkovna_orders.php

```diff
@@ -88,6 +89,63 @@ public function printLabels($orders_id_arr, $format = 'A7 on A4', $offset = '0')
         exit();
     }
 
+    /**
+     * @param int[]|string[] $packetIds
+     * @param Label\Format $format
+     * @param int $offset
+     * @return array
+     */
+    public function printCarrierLabels(array $packetIds, Label\Format $format, $offset = 0)
+    {
+        $errors = [];
+        if (empty($packetIds)) {
+            $errors[] = JText::_('PLG_VMSHIPMENT_PACKETERY_NO_PACKET_TO_PRINT');
+
+            return $errors;
+        }
+
+        $validPacketIds = $this->repository->getExternalCarrierPacketIdsByPacketIds($packetIds);
```

Comment:
Metoda getExternalCarrierPacketIdsByPacketIds() je podle mne zbytečná.  V tomto konkrétním případě není důvod v DB znovu ověřovat,  zda ID zásilek předaných formulářem v DB opravdu existují, protože byly z DB před tím načteny.  Je to zbytečný dotaz do DB a zbytečný kód.

Reply:
Metoda řeší pátý bod zadání.

Reply:
A nebylo by lepší v Order/repository vytvořit metodu getByIds a předávat ID objednávek u kterých chceš vrátit ID zásilky ? Takovou metodu pak můžeš v případě potřeby kdykoli rozšířit i o další sloupce, které budeš potřebovat vrátit,  lze ji používat i na jiných místech.

Reply:
Já nemám k dispozici IDčka objednávek, ale jen zásilek. Proč bych databázi volal dvakrát, když jí mohu volat jednou?

Reply:
celé jsem si to ještě jednou prošel a klidně to nechme.